core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -93,6 +91,9 @@ class SparkHadoopUtil extends Logging {
         if (key.startsWith("spark.hadoop.")) {
           hadoopConf.set(key.substring("spark.hadoop.".length), value)
         }
+        if (key.startsWith("spark.yarn.")) {
```

Comment:
while this would be nice, I agree with @srowen, this feels hacky to me.  We are also adding a bunch of spark.yarn configs to the hadoop configs that aren't needed for anything.

 At this point I don't see any other places to use the variable substitution so perhaps we can special case this one and have a special string to tell spark on yarn  to look in the hadoop config for it.  Like ${hadoop-yarn.resourcemanager.hostname}. We can get the port from spark itself conf.getInt("spark.history.ui.port", 18080)

thoughts on that?